core/src/main/java/com/microsoft/applicationinsights/internal/heartbeat/HeartbeatDefaultPayload.java

```diff
@@ -59,19 +58,17 @@ public static boolean addDefaultPayLoadProvider(HeartBeatPayloadProviderInterfac
   public static Callable<Boolean> populateDefaultPayload(final List<String> disabledFields, final List<String>
       disabledProviders, final HeartBeatProviderInterface provider) {
     return new Callable<Boolean>() {
-
-      volatile boolean populatedFields = false;
       @Override
       public Boolean call() throws Exception {
+        boolean populatedFields = false;
         for (HeartBeatPayloadProviderInterface payloadProvider : defaultPayloadProviders) {
           if (disabledProviders != null && disabledProviders.contains(payloadProvider.getName())) {
 
             // skip any azure specific modules here
             continue;
           }
 
-          boolean fieldsAreSet = payloadProvider.setDefaultPayload(disabledFields, provider).call();
-          populatedFields = populatedFields || fieldsAreSet;
+          populatedFields = populatedFields || payloadProvider.setDefaultPayload(disabledFields, provider).call();
```

Comment:
I don't know if this was intentional, but before it was still executing `setDefaultPayload` on every loop, but now it will not execute `setDefaultPayload` once `populatedFields` is `true`

Reply:
Let me see if there are side effects of setDefaultPayload. It may not need to run every loop

Reply:
```suggestion
          boolean fieldsAreSet = payloadProvider.setDefaultPayload(disabledFields, provider).call();
          populatedFields = populatedFields || fieldsAreSet;
```